libexec/rtld-elf/aarch64/reloc.c

```diff
@@ -536,12 +536,6 @@ reloc_plt(Plt_Entry *plt, int flags, RtldLockState *lockstate)
 		case R_AARCH64_NONE:
 			break;
 #ifdef __CHERI_PURE_CAPABILITY__
-		/*
-		 * XXX Dapeng: The R_MORELLO_RELATIVE case should be removed
-		 * once the whole system is compiled with -cheri-codeptr-relocs,
-		 * for it would then be impossible to see this relocation here.
-		 */
-		case R_MORELLO_RELATIVE:
```

Comment:
This seems premature given that packages are compiled without -cheri-codeptr-relocs?

Reply:
But packages aren’t doing sub-library compartmentalisation? Isn’t this for inter-compartment calls within a library when you use the old relocations?

Reply:
@bsdjhb I think @jrtc27 is right. The compiler already has function-pointer-wrapping merged, so regardless of whether we do sub-library compartmentalisation or not, we won't see `R_MORELLO_RELATIVE` here.

Reply:
But for the sake of supporting `MK_CHERI_CODEPTR_RELOCS=no` we'll still support `R_MORELLO_RELATIVE` for now.